### PR TITLE
fix(resource): cant submit resource request

### LINKED ISSUE #5440: Cant submit Resource Request
Resource Request generation is not working on all access levels.
 The "Submit" button is not responding.

### PR DESCRIPTION
**Pull request by cp-Coder**
# Bug Fix

### <samp>🤖 Generated by Copilot at 1e08fdd</samp>

Refactored the resource creation form and added localization support. The form now uses custom components and the `useTranslation` hook to render the fields and labels in the user's preferred language. The code was also cleaned up and the types and styles were enhanced. The English locale file was updated with the resource-related strings from `Resource.json`.

## Proposed Changes

- Updates `form validation` logic
- Replaces MUI with tailwind components

## Associated Issue

- Fixes #5440

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug/test a new feature.
- [ ] Update product [documentation](https://docs.coronasafe.network/coronasafe-care-documentation/architecture/architecture-and-layering-of-care)
- [ ] Ensure that UI text is kept in I18n files
- [ ] Prep the screenshot or demo video for the changelog entry, and attach it to issue
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### <samp>🤖 Generated by Copilot at 1e08fdd</samp>

*  Remove unused imports from `ResourceCreate.tsx` to clean up the code and avoid unnecessary dependencies ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL4-R13))
*  Add imports of custom form components and translation hook to `ResourceCreate.tsx` to improve the form functionality and appearance ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aR21-R27))
*  Initialize the translation hook with `const { t } = useTranslation();` in `ResourceCreate.tsx` to access the translation function for the form labels and title ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aR88))
*  Move the phone number parsing statement inside the relevant case block in the `validateForm` function in `ResourceCreate.tsx` to avoid unnecessary parsing of other fields ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL137-R134), [link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aR146))
*  Change the `handleChange` function parameter type to `FieldChangeEvent<string>` in `ResourceCreate.tsx` to use the custom type for the form field change event ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL163-R161))
*  Replace the `PageTitle` component prop with the translation function for the page title in `ResourceCreate.tsx` ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL228-R224))
*  Replace the legacy form components with the custom form components and use the translation function for the field labels in `ResourceCreate.tsx` ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL239-R254), [link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-60a064d920f5f0359fb42250988ba67773d6bcf0e5e506b011ece11471ef740aL278-R327))
*  Import the `Resource.json` file to `index.js` in the English locale to include the resource-related translation strings ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-f5a5abfe6011db80ae7ac7d25eb216e76c5fc78a79e46b1b32be04d3853c4271R13))
*  Spread the `Resource` object to the `en` object in the English locale to include the resource-related translation strings ([link](https://github.com/coronasafe/care_fe/pull/5443/files?diff=unified&w=0#diff-f5a5abfe6011db80ae7ac7d25eb216e76c5fc78a79e46b1b32be04d3853c4271R28))

### CODE REVIEW

File: src/Locale/en/Resource.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "create_title": "Create Resource Request",
```
Comment (rithviknishad) [inline src/Locale/en/Resource.json]: @cp-Coder the translation key `create_title` is too generic to be used for a specific `Create Resource Request` and could cause confusion and/or conflicts with other keys.

```suggestion
  "create_resource_request": "Create Resource Request",
```

File: src/Components/Resource/ResourceCreate.tsx [hunk 2]
```diff
@@ -225,7 +221,7 @@ export default function ResourceCreate(props: resourceProps) {
   return (
     <div className="px-2 pb-2">
       <PageTitle
-        title={"Create Resource Request"}
+        title={t("create_title")}
```
Comment (rithviknishad) [inline src/Components/Resource/ResourceCreate.tsx]:
```suggestion
        title={t("create_resource_request")}
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
LGTM

**Comment by khavinshankar** [comment]
@rithviknishad can you rereview this once

**Review by rithviknishad — APPROVED** [review]

**Pull request merged**